utils/cmc.go

```diff
@@ -120,6 +120,9 @@ REFRESH:
 		}
 		if quotemap["quote"] != nil {
 			b, err = json.Marshal(quotemap["quote"])
+			if err != nil {
+				return pricer.coins[coin].price, err
+			}
 			if err := json.Unmarshal(b, &usdmap); err != nil {
 				fmt.Println("error: ", err)
 				return pricer.coins[coin].price, err
```

Comment:
Nice catch, but why is it marshalling and unmarshalling the same data? Is it using marshalling to do a deep copy?

Reply:
long story short, coinmarketcap absolutely butchered they're API, and the index keys for the JSON they send back is dynamic and entirely depends on the total marketcap of the cryptocurrency. It's also not possible to use golang structs to unmarshal this data, because we can't use JSON tags because the index key for the data we want isn't static. So we need to do this absolutely grotesque method of unmarshaling to maps just to get access to the data we want. 

Unless I had a big brain fart, its one of the most poorly design APIs that sends back JSON responses ever.

Reply:
😂